components/ui/data-table.tsx

```diff
@@ -81,6 +96,14 @@ export function DataTable<TData, TValue>({
             </Button>
           </DropdownMenuTrigger>
           <DropdownMenuContent align="end" className="bg-foreground">
+          <DropdownMenuCheckboxItem
```

Comment:
Instead of having the "All" Be an option in dropdown could we make it a Button at the top. Additionally add in a clear button at the top as well so that a user can select all or clear all with a single click. 

Specifically have the buttons be "Select All" and "Clear All". 

- Disable the select all button when all are selected and Disable the Clear All button when None are selected. 

This way we will not need to use state in the way you have above.

Reply:
Reasoning: https://ux.stackexchange.com/questions/43672/should-multi-select-dropdown-box-have-an-all-option

I think interacting with the All as a select option just feels like bad UX to me now so I feel the above approach may be better (something we will find out more in user testing) 

You could have the buttons be top and bottom or side by side, can make a call on that as needed.

Reply:
I agree completely, and implemented the requested changes. See the gif at the top of the PR for details.